### PR TITLE
Allow different translations of "Red Tower"

### PR DESCRIPTION
**Pull request by a1exsh**
Both Sorceress and Warlock have a building named "Red Tower".  In some
translations, e.g. Russian "Buka" version, they are translated differently,
which makes it a bit nicer.

In order to support the same we have to use distinct translatable strings.

### CODE REVIEW

**Comment by a1exsh** [comment]
Not updating translation files yet to avoid conflicts, before this is merged: https://github.com/ihhub/fheroes2/pull/4971

**Comment by ihhub** [comment]
Hi @a1exsh , you can update now.

**Comment by a1exsh** [comment]
@ihhub updated, please check.  An example distinct translation is provided for `ru.po`.

**Comment by ihhub** [comment]
Hi @zenseii , @Arthusppp, @LeHerosInconnu and @shprotru , could you please help with translations for other languages?

**Review by ihhub — APPROVED** [review]

File: files/lang/nb.po [hunk 1]
```diff
@@ -2063,7 +2063,8 @@ msgstr "Oppg. steinring"
 msgid "Fenced Meadow"
 msgstr "Inngjerdet slette"
 
-msgid "Red Tower"
+#, fuzzy
+msgid "sorceress|Red Tower"
 msgstr "Rødt tårn"
```
Comment (zenseii) [inline files/lang/nb.po:2068]:
```suggestion
msgstr "Ildtårn"
```

File: src/fheroes2/castle/castle_building_info.cpp [hunk 2]
```diff
@@ -528,7 +528,7 @@ namespace
         case DWELLING_MONSTER5:
             return _( "Fenced Meadow" );
         case DWELLING_MONSTER6:
-            return _( "Red Tower" );
+            return _( "sorceress|Red Tower" );
```
Comment (zenseii) [inline src/fheroes2/castle/castle_building_info.cpp:531]: Do we want to change this in English as well then? For me I don't mind either leaving it as is or changing it.

**Review by zenseii — COMMENTED** [review]
Hi, @a1exsh.

I've left a couple of comments. When you have the time these can be looked at.

**Comment by LeHerosInconnu** [comment]
Hello @a1exsh, @ihhub and @zenseii,

> Hi @zenseii , @Arthusppp, @LeHerosInconnu and @shprotru , could you please help with translations for other languages?

I do not find it disturbing that buildings have the same name if they are not located in the same castle, especially since there is only one case.
Anyway here are some ideas for the red tower of the sorceress castle:

- Tour écarlate
- Tour de feu
- Tour de flammes
- Tour flamboyante
- Tour ardente
- Tour de braises
- Tour du Soleil
- Tour de sang
- Tour éternelle
- Tour Eiffel :rofl:

**Comment by Arthusppp** [comment]
> I do not find it disturbing that buildings have the same name if they are not located in the same castle, especially since there is only one case.

Agree, in Polish, we have the "same situation". Nevertheless, I totally understand that in different languages it can be a different situation. Therefore, I have translated both "Red Towers" using the same name. :)

**Comment by ihhub** [comment]
@a1exsh , thank you for these changes!

I merged this pull request to proceed further with translations since the author of this PR is busy at the moment.

**Pull request merged**